### ISSUE TITLE
Instance layout conflict false positive with multiple inheritance from C implementations of `asyncio.Task` and `asyncio.Future`

### ISSUE THREAD

**Issue by jonathandung** [astral-sh/ty#3466]
### Summary

```python
import asyncio
print(issubclass(asyncio.Task, asyncio.Future)) # True
class Future(asyncio.Future): ...
class Task(asyncio.Task): ...
class SubClass(Task, Future): ... # Succeeds at runtime; ty flags instance-layout-conflict, claiming `TypeError` will be thrown
```

### Version

ty 0.0.35 (bc12d1cad 2026-05-10)

**Comment by charliermarsh** [comment]
Closed by https://github.com/astral-sh/ruff/pull/25178.
